hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -215,8 +219,8 @@ abstract class BaseMergeOnReadSnapshotRelation(sqlContext: SQLContext,
       HoodieFileIndex.convertFilterForTimestampKeyGenerator(metaClient, partitionFilters)
 
     if (globPaths.isEmpty) {
-      val fileSlices = fileIndex.listFileSlices(convertedPartitionFilters)
-      buildSplits(fileSlices.values.flatten.toSeq)
+      val fileSlices = fileIndex.filterFileSlices(dataFilters, fileIndex.getFileSlicesForPrunedPartitions(convertedPartitionFilters)).flatMap(s => s._2)
```

Comment:
I see both, MOR snapshot relation and base file relation (HoodieFileIndex.listFiles()), are invoking this. So, might as well move fileIndex.getFileSlicesForPrunedPartitions into fileIndex.filterFileSlices if all the callers are anyway going to invoke the other method

Reply:
Addressed